### ISSUE TITLE
import/extensions doesn't work for require('./file-without-extension')

### ISSUE THREAD

**Issue by balupton** [benmosher/eslint-plugin-import#1230]
See reproduction details at https://github.com/balupton/eslint-issue-import-extensions

---

get started:

```
git clone https://github.com/balupton/eslint-issue-import-extensions.git
cd eslint-issue-import-extensions
npm install
```

run the test:

```
npm test
```

notice that there was no complaint about:

``` js
const result = require('./module')
```

inside `./source/index.js`

**Comment by ljharb** [comment]
The first bullet point seems like a major flaw in both vscode and typescript, since omitting the extension is a best practice. Hopefully issues have been filed for those.

Separately, we should lint import and require statements both whenever possible, for every rule - so this is a good fix to make.

**Comment by balupton** [comment]
> Separately, we should lint import and require statements both whenever possible, for every rule - so this is a good fix to make.

agreed, thanks, looking forward to it

**Comment by ljqx** [comment]
It seems better to have a helper to generate callbacks for `ImportDeclaration`/`ExportNamedDeclaration` and static require `CallExpression` together?
Sth like following in rules:
```js
module.exports = {
  create(context) {
    return addForGenericImport({
      ... // other callbacks
    }, (source, node) => {
      // callbacks for ImportDeclaration, static require CallExpression, ExportNamedDeclaration, etc
    }),
  }
}
```
Actually most rules miss `ExportNamedDeclaration`, and some rules miss static require `CallExpression`, not only `import/extensions`.

How do you think? @ljharb

**Comment by ljharb** [comment]
A commonly used helper seems like a good idea.
